synapse/glob.py

```diff
@@ -2,14 +2,8 @@
 
 lock = threading.RLock()
 
-# A global / default PkiStor used by getUserPki()
-pki = None
+import synapse.lib.sched as s_sched
+import synapse.lib.threads as s_threads
 
-# A global Plex instance for managing socket i/o.
-plex = None
-
-# A global Sched instance for maint routines
-sched = None
-
-# A host info dictionary for "thishost" API
-hostinfo = None
+pool = s_threads.Pool()
```

Comment:
The pool ``__ini__`` sig is the following:
``
def __init__(self, size=3, maxsize=None):
``

This results in a pool of global pool of three threads max to run things from.

Reply:
This seems like a small pool for a global thread pool

Reply:
For context, the ThreadPoolExecutor from concurrent.futures uses the following for the default number of max workers ``max_workers = (os.cpu_count() or 1) * 5``

Reply:
Actually, it ends up with a thread pool of *minimum* size 3, with no maximum size, which will keep >size threads alive until there is not more work to do, but then let them die off ( to a minimum of size, so the next requests that come in dont have to fire a new one ).  I was considering setting a max, but most of our use cases so far haven't used one.